sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -1870,4 +1870,66 @@ class SQLQuerySuite extends QueryTest with SharedSQLContext {
       assert(sampled.count() == sampledOdd.count() + sampledEven.count())
     }
   }
+
+  test("Struct Star Expansion") {
+    checkAnswer(
+      sql("select struct(a, b) as record from testData2").select($"record.a", $"record.b"),
```

Comment:
you can make it more dataframe style like `testData2.select(struct("a", "b").as("record"))`